### PR TITLE
[linux-port] Correct problem initializations

### PR DESCRIPTION
**Pull request by pow2clk**
Regardless of the order of variables placed in a constructor
member initializer list, the variables are initialized in the
order they are listed in the class or struct. This can cause
issues where the assignment of one variable depends on the other.
For this reason, a warning is produced if the member variables
are listed in a different order. This change moves the member
initializer lists into the order they were being initialized.
Fixes 23 clang and 24 gcc warnings.

Initialization of members of a larger aggregate type may benefit
from braces to clarify which variables are expected to go where.
At the same time, scalars should not be so segregated. This places
or removes such braces wherever appropriate.
Fixes 144 clang warnings

Contributes to https://github.com/google/DirectXShaderCompiler/issues/206

### CODE REVIEW

File: lib/HLSL/HLOperationLower.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ struct HLObjectOperationLowerHelper {
     // Add invalid first to avoid dead loop.
     HandleMetaMap[Handle] = {DXIL::ResourceClass::Invalid,
                              DXIL::ResourceKind::Invalid,
-                             StructType::get(Type::getVoidTy(HLM.GetCtx()))};
+                             StructType::get(Type::getVoidTy(HLM.GetCtx()), nullptr)};
```
Comment (pow2clk) [inline lib/HLSL/HLOperationLower.cpp]: This was an interesting one. getVoidTy is marked as a sentinal, which means if you don't provide a null in the last spot, the compiler warns you something bad(tm) could happen. I wasn't aware of that feature before.

File: lib/HLSL/DxilModule.cpp [hunk 2]
```diff
@@ -83,7 +69,22 @@ DxilModule::DxilModule(Module *pModule)
...
+, m_pDebugInfoFinder(nullptr)
+, m_pSM(nullptr)
+, m_DxilMajor(DXIL::kDxilMajor)
+, m_DxilMinor(DXIL::kDxilMinor)
+, m_ValMajor(1)
+, m_ValMinor(0)
+, m_pOP(llvm::make_unique<OP>(pModule->getContext(), pModule))
+, m_pTypeSystem(llvm::make_unique<DxilTypeSystem>(pModule))
+, m_pViewIdState(llvm::make_unique<DxilViewIdState>(this)) {
+
```
Comment (pow2clk) [inline lib/HLSL/DxilModule.cpp]: Feel free to make sure, but this should just be a resorting. So no need to look for deeper meaning if you want to trust me.

Comment (ehsannas) [reply lib/HLSL/DxilModule.cpp]: LGTM!

File: lib/HLSL/HLOperationLower.cpp (hunk 1 above)
Comment (ehsannas) [reply lib/HLSL/HLOperationLower.cpp]: Curious about why this is.

**Pull request merged**